esp-hal/src/i2c.rs

```diff
@@ -2240,44 +2186,37 @@ where
     I: Iterator<Item = &'a COMD>,
 {
     let cmd = cmd_iterator.next().ok_or(Error::CommandNrExceeded)?;
-    unsafe {
-        match command {
-            Command::Start => {
-                cmd.write(|w| w.opcode().rstart());
-            }
-            Command::Stop => {
-                cmd.write(|w| w.opcode().stop());
-            }
-            Command::End => {
-                cmd.write(|w| w.opcode().end());
-            }
-            Command::Write {
-                ack_exp,
-                ack_check_en,
-                length,
-            } => {
-                cmd.write(|w| {
-                    w.opcode().write();
-                    w.ack_exp().bit(ack_exp == Ack::Nack);
-                    w.ack_check_en().bit(ack_check_en);
-                    w.byte_num().bits(length);
-                    w
-                });
-            }
-            Command::Read { ack_value, length } => {
-                cmd.write(|w| {
-                    w.opcode().read();
-                    w.ack_value().bit(ack_value == Ack::Nack);
-                    w.byte_num().bits(length);
-                    w
-                });
-            }
+
+    match command {
```

Comment:
I'm a little surprised you didn't move the `match` into the `cmd.write(|w| .....)` lambda. :stuck_out_tongue:

Reply:
oops